src/main/app/initialize.ts

```diff
@@ -291,6 +297,24 @@ function initializeInterCommunicationEventListeners() {
 }
 
 async function initializeAfterAppReady() {
+    protocol.handle('mattermost-desktop', (request: Request) => {
+        const url = parseURL(request.url);
+        if (!url) {
+            return new Response('bad', {status: 400});
+        }
+
+        // Including this snippet from the handler docs to check for path traversal
+        // https://www.electronjs.org/docs/latest/api/protocol#protocolhandlescheme-handler
+        const pathToServe = path.join(app.getAppPath(), 'renderer', url.pathname);
```

Comment:
It should be pretty specific to this use case since I'm not really allowing for any other arbitrary paths to be loaded, outside of preload scripts (but those are all included in the bundle and cannot be overridden via normal app usage)